### PR TITLE
Enabling focus and accessible keyboard navigation on Spearmint.

### PR DESCRIPTION
**Pull request by bielchristo**
# Enabling focus and accessible keyboard navigation on Spearmint.
<hr>

This PR reflects changes in styling and focus attribute of rendered components which allow users to navigate on Spearmint with the use of the keyboard only if they so desire. Primary buttons and Input fields are now visibly outlined in dark-blue upon loading throughout the application, which enables for a more accessible experience to all users in compliance with the WCAG standards.  
Additionally, the delete functionality of the It and Describe blocks for the React and Accessibility portions is now working and accessible by the keyboard.
<hr>

## Navigation example:

Hnet-image (2)

## Deleting via keyboard:

Hnet-image (3)

<hr>

## A few important changes to be aware of:

* Implemented '`autofocus`' on each page of our app
* Removed instances of "`useEffect`" which were previously defaulting focus to describe blocks.
* Used SCSS to give focused items a dark blue border.
* Added "`onKeyPress`" functionality to `<i>` tags that handled delete Describe and It blocks but were inaccessible through keystrokes.

<hr>
 
## This PR alters 20 files
```

        src/assets/stylesheets/reset.scss
        src/components/AccTestComponent/DescribeRenderer/DescribeRenderer.tsx
        src/components/EndpointTestComponent/Endpoint.tsx
        src/components/Modals/ExportFileModal.module.scss
        src/components/OpenFolder/OpenFolderButton.module.scss
        src/components/ReactTestComponent/DescribeRenderer/DescribeRenderer.jsx
        src/components/SearchInput/SearchInput.jsx
        src/components/TestCase/HooksTestCase.tsx
        src/components/TestCase/ReduxTestCase.tsx
        src/components/TestMenu/AccTestMenu.tsx
        src/components/TestMenu/EndpointTestMenu.tsx
        src/components/TestMenu/HooksTestMenu.tsx
        src/components/TestMenu/PuppeteerTestMenu.tsx
        src/components/TestMenu/ReactTestMenu.jsx
        src/components/TestMenu/ReduxTestMenu.tsx
        src/components/TestMenu/TestMenu.module.scss
        src/pages/ProjectLoader/ProjectLoader.jsx
        src/pages/TestFile/TestFile.jsx
        src/components/ReactTestComponent/ItRenderer/ItRenderer.jsx
        src/components/AccTestComponent/ItRenderer/ItRenderer.tsx
```

Co-Authored by Sharon Zhu - 76144878+sharon-zhu@users.noreply.github.com

### CODE REVIEW

**Review by annieshinn — APPROVED** [review]
Hi Gabi!

To sum up your changes, it looks like you:
1. Implemented 'autofocus' on each page of our app
2. Removed instances of "useEffect" (are all instances removed, or just some?)
3. Used SCSS to give focused items a dark blue border.

Please note the few comments made throughout. They are not very crucial, so I'm going to go ahead and approve. Can't wait to discuss in the morning!

Thanks,
Annie

**Review by annieshinn — APPROVED** [review]
Hey Gabi! The changes look good. Thanks for implementing changes we discussed. I still see some lingering Dark-Green in OpenFolderButton's scss file. Please fix. Otherwise, approved!

**Review by sharon-zhu — APPROVED** [review]
Looking good here. Excited for this feature! Thanks.

File: src/components/AccTestComponent/DescribeRenderer/DescribeRenderer.tsx [hunk 1]
```diff
@@ -13,12 +13,8 @@ const DescribeRenderer = ({
   handleChangeDescribeText,
   handleChangeItStatementText,
   type,
-}): AccTestCaseState => {
-  const testDescription = useRef<HTMLInputElement>(null!);
+}) => {
```
Comment (annieshinn) [inline src/components/AccTestComponent/DescribeRenderer/DescribeRenderer.tsx:16]: Do we want the typescript back?

**Review by annieshinn — APPROVED** [review]
Approved, but please explain the reverting typescript at code review!

File: src/components/AccTestComponent/DescribeRenderer/DescribeRenderer.tsx (hunk 1 above)
Comment (annieshinn) [reply src/components/AccTestComponent/DescribeRenderer/DescribeRenderer.tsx:16]: Discussed.

**Pull request merged**